### PR TITLE
Add Buildkite metadata/links to scans

### PR DESCRIPTION
**Pull request by ZacSweers**
This is done eyeballing the available variabels here: https://buildkite.com/docs/pipelines/environment-variables#buildkite-environment-variables

I've also set it up to link both the PR (if available) and the original commit sha.

I wanted to add a test for some of the URL handling, hope that's ok!

Resolves https://github.com/gradle/common-custom-user-data-gradle-plugin/issues/97

### CODE REVIEW

File: src/main/java/com/gradle/Utils.java [hunk 1]
```diff
@@ -210,6 +210,26 @@ private static boolean isGradleNewerThan(String version) {
         return GradleVersion.current().compareTo(GradleVersion.version(version)) >= 0;
     }
 
+    /**
+     * Construct a repo URL from a git URL in the format of <code>git://github.com/acme-inc/my-project.git</code>. If
+     * the URL cannot be parsed, {@link Optional#empty()} is returned.
+     */
+    static Optional<String> extractRepoUrl(String gitUrl) {
```
Comment (runningcode) [inline src/main/java/com/gradle/Utils.java:217]: This is similar to the `toWebRepoUri` function in `CustomBuildScanEnhancements` can we re-use that function?

**Review by runningcode — COMMENTED** [review]
Thanks for the contribution, Zac. I left a comment.

Could you please also take care of the DCO check and sign the commits?

File: src/main/java/com/gradle/Utils.java (hunk 1 above)
Comment (ZacSweers) [reply src/main/java/com/gradle/Utils.java:217]: Missed that, let me take a look

**Comment by ZacSweers** [comment]
I don't fully understand the DCO check. It wants me to just add "signed off by" to the ends of my commit messages or it wants me to set up full commit signing too? If the latter, I'm not sure how to even do that retroactively to commits 🤔

File: src/main/java/com/gradle/Utils.java (hunk 1 above)
Comment (ZacSweers) [reply src/main/java/com/gradle/Utils.java]: Refactored it out to utils and kept the test if that looks alright to you. Also updated it to support `git://` schems. [e60c70f](https://github.com/gradle/common-custom-user-data-gradle-plugin/pull/169/commits/e60c70f4ca8b432640838c413a9049f57e6ab714)

File: build.gradle [hunk 2]
```diff
@@ -20,6 +20,7 @@ repositories {
 
 dependencies {
     compileOnly 'com.gradle:gradle-enterprise-gradle-plugin:3.13'
+    testImplementation 'junit:junit:4.13.2'
```
Comment (runningcode) [inline build.gradle:23]: Can we use Junit 5?

Comment (ZacSweers) [reply build.gradle:23]: I've never used it and dunno how

File: build.gradle [hunk 3]
```diff
@@ -20,6 +20,7 @@ repositories {
 
 dependencies {
     compileOnly 'com.gradle:gradle-enterprise-gradle-plugin:3.13.1'
+    testImplementation 'junit:junit:4.13.2'
```
Comment (runningcode) [inline build.gradle:23]:
```suggestion
    testImplementation(platform('org.junit:junit-bom:5.9.3'))
	testImplementation('org.junit.jupiter:junit-jupiter')
```

You may also need:

```
test {
	useJUnitPlatform()
}
```

Comment (ZacSweers) [reply build.gradle]: [9580449](https://github.com/gradle/common-custom-user-data-gradle-plugin/pull/169/commits/95804491c3badf954318461205fe63e77a96c119)

File: build.gradle (hunk 2 above)
Comment (ZacSweers) [reply build.gradle]: [9580449](https://github.com/gradle/common-custom-user-data-gradle-plugin/pull/169/commits/95804491c3badf954318461205fe63e77a96c119)

**Comment by ZacSweers** [comment]
Hey folks, I'm a little confused on how to proceed here. CI appears to be passing, but I'm being told I need to go back and sign every commit retroactively. This is difficult to set up on my work laptop and the only repo I've ever contributed to that requires this. Would it be easier to just... squash all my commits into one of the verified commits @ribafish pushed to this branch?

**Comment by ZacSweers** [comment]
Thanks!

**Pull request merged**